server/src/main/java/org/elasticsearch/common/joda/DateMathParser.java

```diff
@@ -44,7 +46,20 @@ public DateMathParser(FormatDateTimeFormatter dateTimeFormatter) {
     }
 
     public long parse(String text, LongSupplier now) {
-        return parse(text, now, false, null);
+        return parse(text, now, false, (DateTimeZone) null);
```

Comment:
Maybe it would be good (for ease of removal in the future) to flip around the joda impl below, so that it is the wrapper method, and the core parse method (which handles no time zone) is the same that handles ZoneId? This could be a follow up.